### ISSUE TITLE
Wrong render result while processing multi-meshes: using join_meshes_as_scene([mesh0, mesh1,mesh2,...])

### ISSUE THREAD

**Issue by Frank-Dz** [facebookresearch/pytorch3d#397]
![Selection_057](https://user-images.githubusercontent.com/55009565/95604689-877fdb00-0a8a-11eb-8a07-7b1da44c0d5e.png)

As shown in the above picture, I used
```
            mesh = join_meshes_as_scene([inside_body_mesh, Shirt_mesh, Pant_mesh])
```
To render a scene with multi meshes that have a clear occlusion.
Yet, seems like that the result can not reflect the occlusion relation of the meshes correctly (**The pant get transparent**).

Below is the result in Meshlab. See the leg part.
![image](https://user-images.githubusercontent.com/55009565/95605873-22c58000-0a8c-11eb-8591-0f98ceb448ff.png)

The pant is not transparent here.

I am using 
```
SoftPhongShader
```

**Comment by Frank-Dz** [comment]
To make the reproduction easier, attached is the test_data2:
[test_data2.zip](https://github.com/facebookresearch/pytorch3d/files/5356080/test_data2.zip)


Thank you for you kind help!

**Comment by nikhilaravi** [comment]
@Frank-Dz I was able to reproduce the issue and resolve it. The transparency is happening because of the setting of the blending parameter `gamma`. The `SoftPhongShader` has an additional input arg called `blend_params` which has two settings :
- `sigma` controls the edge blur
- `gamma` controls the opacity

The default value of `gamma` is `1e-4` which gives the result that you're seeing. If you decrease it to `1e-6` you should see the result below:

<img width="485" alt="Screen Shot 2020-10-09 at 2 25 19 PM" src="https://user-images.githubusercontent.com/5912647/95632996-a1460180-0a3b-11eb-9881-efa981f41a3b.png">


This is an example of how to use the `BlendParams`. 
```
from pytorch3d.renderer import BlendParams
blend_params = BlendParams(gamma=1e-6)
renderer = MeshRenderer(
      rasterizer=MeshRasterizer(
      cameras=cameras, 
      raster_settings=raster_settings
    ),
    shader=SoftPhongShader(
      device=device, 
      cameras=cameras,
      lights=lights,
      blend_params=blend_params
    )
)

**Comment by Frank-Dz** [comment]
@nikhilaravi Thank you!  Yes, it is.

Best,
Frank

**Comment by Frank-Dz** [comment]
@nikhilaravi 
Plus, As for the issue:
https://github.com/facebookresearch/pytorch3d/issues/396

This is because the render generates the value which can be larger than 1.

To make a comparison, I also tried DIRT(it can render it). And check the model in MeshLab:
![image](https://user-images.githubusercontent.com/55009565/95645837-77e9ac00-0af5-11eb-84ad-def815379d7c.png)

And, if we change the render setting, e.g, set the material's shininess to a different value, this problem can be solved. But when I test on other models, this problem appears again.

Is there any specific setting or requirements for the input?
For example, the RGB value of the texture image should satisfy some requirements?
According to the Documentation, I did not find such kind of requirements.





[0-model.zip](https://github.com/facebookresearch/pytorch3d/files/5358463/0-model.zip)


```
# Imports
import torch
import os
import time
import matplotlib.pyplot as plt
from skimage.io import imread
import numpy as np
import matplotlib
matplotlib.use('Agg')
# Util function for loading meshes
from pytorch3d.io import load_objs_as_meshes, load_obj
# Data structures and functions for rendering
from pytorch3d.structures import Meshes
from pytorch3d.renderer import BlendParams

# from pytorch3d.visualization import AxisArgs, plot_meshes
from pytorch3d.renderer import (
    look_at_view_transform,
    FoVPerspectiveCameras, 
    PointLights, 
    DirectionalLights, 
    Materials, 
    RasterizationSettings, 
    MeshRenderer, 
    MeshRasterizer,  
    SoftPhongShader,
    HardPhongShader,
    TexturesUV,
    TexturesVertex
)
from pytorch3d.structures.meshes import (
    Meshes,
    join_meshes_as_batch,
    join_meshes_as_scene,
)
import sys
import os
sys.path.append(os.path.abspath(''))
if torch.cuda.is_available():
    device = torch.device("cuda:0")
    torch.cuda.set_device(device)
else:
    device = torch.device("cpu")

# Set paths
DATA_DIR = "./data"

if __name__ == "__main__":
    '''
    First we set up render
    '''
    batch_size = 8
    azim = torch.linspace(0.0, 325.0, batch_size)
    blend_params = BlendParams(gamma=1e-6)

    R, T = look_at_view_transform(dist=1.8, azim = azim, elev =5, at=((0,-0.25, 0),))
    cameras = FoVPerspectiveCameras(device=device, R=R, T=T)
    raster_settings = RasterizationSettings(
    image_size=512, 
    blur_radius=0.0, 
    faces_per_pixel=2,  # more faces will caused GPU out of Memory.
    )
    lights = PointLights(device=device,ambient_color =((0.55, 0.55, 0.55),))
    lights.location = torch.tensor([2.0 , 300.0, 2.0], device=device)[None]
    # lights.location = torch.tensor([[0.0, 0.0, -3.0]], device=device)
    renderer = MeshRenderer(
        rasterizer=MeshRasterizer(
        cameras=cameras, 
        raster_settings=raster_settings
        ),
        shader=SoftPhongShader(
        device=device, 
        cameras=cameras,
        lights=lights,
        blend_params=blend_params
        )
    )
    materials = Materials(
    device=device,
    shininess=1,
    )
    
    
    obj_filename0 = '0.obj'
    # obj_filename1 = '1.obj'
    # obj_filename2 = '2.obj'
    mesh0 = load_objs_as_meshes([obj_filename0], device=device)
    # mesh1 = load_objs_as_meshes([obj_filename1], device=device)
    # mesh2 = load_objs_as_meshes([obj_filename2], device=device)
    texture_image0=mesh0.textures.maps_padded()
    # texture_image1=mesh1.textures.maps_padded()
    # texture_image2=mesh2.textures.maps_padded()
    # mesh = join_meshes_as_scene([mesh0, mesh1,mesh2])
            

    meshes = mesh0.extend(batch_size)
    # meshes = mesh0
    images = renderer(meshes, cameras=cameras, lights=lights,materials = materials)
    counter = 1

    

    for i in images:
        plt.figure(figsize=(512, 512))
        aa = i[..., :3].cpu().numpy()
        for index1 in range(512):
            for index2 in range(512):
                for index3 in range(3):
                    if (aa[index1,index2,index3] >=0 and aa[index1,index2,index3] <=1):
                        continue
                    else:
                        print(aa[index1,index2,index3])
                        print('at this time: ',counter)
                        input("stop")
        plt.imsave('view%d.png'%counter, aa)
        # input("stop")
        plt.grid("off");
        plt.axis("off");
        plt.close()
        counter = counter + 1

    

```
